core/gear_log/file_handle.ex

```diff
@@ -20,56 +20,57 @@ defmodule AntikytheraCore.GearLog.FileHandle do
     end
   end
 
-  @opaque t :: {Path.t, Time.t, File.io_device}
+  @opaque t :: {Path.t, Time.t, File.io_device, boolean}
 
-  defun open(file_path :: Path.t) :: t do
+  defun open(file_path :: Path.t, opts \\ []) :: t do
+    write_to_terminal = Keyword.get(opts, :write_to_terminal, determine_write_to_terminal())
     :ok = File.mkdir_p(Path.dirname(file_path))
     if File.exists?(file_path) do
       rename(file_path)
     end
-    {file_path, Time.now(), open_file(file_path)}
+    {file_path, Time.now(), open_file(file_path), write_to_terminal}
   end
 
-  defun write({file_path, last_checked_at, io_device} = handle   :: t,
-              {now, _, _, _}                          = gear_log :: Message.t) :: {:kept_open | :rotated, t} do
+  defun write({file_path, last_checked_at, io_device, write_to_terminal} = handle   :: t,
+              {now, _, _, _}                                             = gear_log :: Message.t) :: {:kept_open | :rotated, t} do
     if SizeCheck.check_now?(now, last_checked_at) do
       if SizeCheck.exceeds_limit?(file_path) do
-        {_, _, new_io_device} = new_handle = rotate(handle)
-        do_write(new_io_device, gear_log)
+        {_, _, new_io_device, _} = new_handle = rotate(handle)
+        do_write(new_io_device, gear_log, write_to_terminal)
         {:rotated, new_handle}
       else
-        do_write(io_device, gear_log)
-        {:kept_open, {file_path, now, io_device}}
+        do_write(io_device, gear_log, write_to_terminal)
+        {:kept_open, {file_path, now, io_device, write_to_terminal}}
       end
     else
-      do_write(io_device, gear_log)
+      do_write(io_device, gear_log, write_to_terminal)
       {:kept_open, handle}
     end
   end
 
-  defunp do_write(io_device :: :file.io_device, {time, level, context_id, msg} :: Message.t) :: :ok do
+  defunp do_write(io_device :: :file.io_device, {time, level, context_id, msg} :: Message.t, write_to_terminal :: boolean) :: :ok do
     prefix = log_prefix(time, level, context_id)
     formatted_lines_str = String.split(msg, "\n", trim: true)
       |> Enum.reduce("", fn(s, acc) -> acc <> prefix <> s <> "\n" end)
     :ok = IO.binwrite(io_device, formatted_lines_str)
-    write_debug_log(level, formatted_lines_str)
+    if write_to_terminal, do: write_debug_log(level, formatted_lines_str)
```

Comment:
```suggestion
    if write_to_terminal, do: write_debug_log(level, formatted_lines_str), else: :ok
```

I think that returning `:ok` or `nil` is not a problem for now because the returned value of `do_write/2` is not used anywhere.
But to retain the previous behavior, I think returning`:ok` always is preferable.

Reply:
thanks!